### DISCUSSION TITLE
Some general topics that popped into my head

### DISCUSSION THREAD

**Discussion by Wachiwi** [open-sauced/open-sauced#866] [General] [answered]
First things first - awesome idea and implementation so far with that app. I initially loved the approach and visualization to that problem of getting into open source. I read through some issues and was ready to get started but had my troubles with doing so. I know what was missing code wise and how I needed to implement it but I couldn't. Here perhaps I wanted to discuss some questions and ideas in general that popped into my head browsing the source code at that time.

### Topic 1:  What about a "store"?

It is some time ago since I have done React - lately, it was more using Vue and Svelte - so I don't know all the hooks stuff in detail. There was an issue regarding usability (#832, #862) and I thought that an enriched store/state management would be very helpful here. 

The question here would be: Is there anything like that already in place and I simply didn't see it?

### Topic 2: Refactoring UI to focus key elements

On the dashboard, for example, the most important point is what repositories are on your list. but it is below the intro. By moving it up and only providing an onboarding process if there are no repositories on the list. Also, it could be an improvement giving the repositories only 2/3 of the width and 1/3 would be something like recommended repositories.

The question here would be: Is this something that makes sense to you? Is there already a "vision" for that defined aka in someone's head?

### Topic 3: Components don't need to be components

I know everything is a component technical wise but semantically I personally more to separate components "semantically", if I can say so. What I mean with that is that there are components, views and pages (as in Next.js/Nuxt etc.). What do you think about doing it here as well? This could lead in my opinion to a simpler, more maintainable structure that is easy for contributors to understand and follow.

---

Now it's your turn - let's start a discussion. 

PS: No bashing intended, loving the idea & approach and I only want to help to push the idea forward.

**Comment by bdougie** [comment, marked as answer]
@Wachiwi thanks for opening this, sorry it this long for a response. These are great questions that I have been thinking about. 

This project pre-dates the existence of a lot of these questions and follows some old patterns in places. 

> Topic 1: What about a "store"?

At one time this project had redux, but is removed in favor of useContext. At the moment the only place it is used is to store the goal (open-sauced-goals) id. It has been a busy few months, but I hope to evolve this project and build something a little more sophisticated. I hope to start planning the 2021 roadmap closer to the holidays.

> Topic 2: Refactoring UI to focus key elements

The UI is a bit of a mess and needs help everywhere. I am interested in what you mean by refocuing on key elements. I also consider just opting for Tailwind, but want to be slow to making a decision like that since it will effect the project long term (good and bad).

> Topic 3: Components don't need to be components

100% agree and learning about Next in other projects. Happy to explore this more.

**Reply by 0-vortex** [reply]
+1 on next
